



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="Documentation for RNAsamba, a coding potential assessment tool">
      
      
        <link rel="canonical" href="https://apcamargo.github.io/RNAsamba/usage/">
      
      
        <meta name="author" content="Antonio Camargo">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>Using RNAsamba locally - RNAsamba documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application.adb8469c.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-palette.a8b3c06d.css">
      
      
        
        
        <meta name="theme-color" content="">
      
    
    
      <script src="../assets/javascripts/modernizr.86422ebf.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:300,400,400i,700%7CSource+Code+Pro&display=fallback">
        <style>body,input{font-family:"Lato","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Source Code Pro","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../extra.css">
    
    
      
    
    
  </head>
  
    
    
    <body dir="ltr" data-md-color-primary="white" data-md-color-accent="green">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#using-rnasamba-locally" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://apcamargo.github.io/RNAsamba/" title="RNAsamba documentation" aria-label="RNAsamba documentation" class="md-header-nav__button md-logo">
          
            <i class="md-icon">book</i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              RNAsamba documentation
            </span>
            <span class="md-header-nav__topic">
              
                Using RNAsamba locally
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" aria-label="search" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/apcamargo/RNAsamba/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    apcamargo/RNAsamba
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="https://apcamargo.github.io/RNAsamba/" title="RNAsamba documentation" class="md-nav__button md-logo">
      
        <i class="md-icon">book</i>
      
    </a>
    RNAsamba documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/apcamargo/RNAsamba/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    apcamargo/RNAsamba
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../theory/" title="Theory" class="md-nav__link">
      Theory
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Using RNAsamba locally
      </label>
    
    <a href="./" title="Using RNAsamba locally" class="md-nav__link md-nav__link--active">
      Using RNAsamba locally
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#download-the-pre-trained-models" class="md-nav__link">
    Download the pre-trained models
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rnasamba-train" class="md-nav__link">
    rnasamba train
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rnasamba-classify" class="md-nav__link">
    rnasamba classify
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examples" class="md-nav__link">
    Examples
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-the-docker-image" class="md-nav__link">
    Using the Docker image
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#download-the-pre-trained-models" class="md-nav__link">
    Download the pre-trained models
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rnasamba-train" class="md-nav__link">
    rnasamba train
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rnasamba-classify" class="md-nav__link">
    rnasamba classify
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examples" class="md-nav__link">
    Examples
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-the-docker-image" class="md-nav__link">
    Using the Docker image
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/apcamargo/RNAsamba/edit/master/docs/usage.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="using-rnasamba-locally">Using RNAsamba locally</h1>
<h2 id="download-the-pre-trained-models">Download the pre-trained models</h2>
<div class="admonition question">
<p class="admonition-title">What is a weights file?</p>
<p>When a neural network classification model is trained the information it learns is numerically encoded in its weights. In order to use RNAsamba to classify your own sequences, you need to load weights obtained in a previous training iteration.</p>
</div>
<p>We provide two HDF5 files containing the weights of classification models trained with human trancript sequences. The first model (<code>full_length_weights.hdf5</code>) was trained with full-length transcripts and can be used in datasets comprised mostly or exclusively of complete transcript sequences. The second model (<code>partial_length_weights.hdf5</code>) was trained with both complete and truncated transcripts and is prefered in cases where there is a significant fraction of partial-length sequences, such as transcriptomes assembled using <em>de novo</em> approaches. Both models achieve high classification performance in transcripts from a variety of different species (see our <a href="https://academic.oup.com/nargab/article/2/1/lqz024/5701461">article</a>). <strong>For most cases, we reccomend using <code>full_length_weights.hdf5</code>, unless you have a reason to suspect that your data contain a significative amount of truncated transcripts.</strong></p>
<p><center>
  <button onclick="location.href='https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/full_length_weights.hdf5'" class="pure-material-button-contained">Full-length transcripts</button>
  <button onclick="location.href='https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/partial_length_weights.hdf5'" class="pure-material-button-contained">Partial-length transcripts</button>
</center></p>
<p>Alternatively, you can download the weight files in an command-line interface by executing the following commands:</p>
<div class="codehilite"><pre><span></span><code><span class="err">curl -O https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/full_length_weights.hdf5</span>
<span class="err">curl -O https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/partial_length_weights.hdf5</span>
</code></pre></div>


<p>In case you want to train your own model, you should follow the steps described in the <a href="#rnasamba-train"><code>rnasamba train</code></a> section.</p>
<h2 id="rnasamba-train"><code>rnasamba train</code></h2>
<p><code>rnasamba train</code> is the command for training a new classification model from a training dataset and saving the network weights into an HDF5 file. The user can specify the batch size (<code>--batch_size</code>) and the number of training epochs (<code>--epochs</code>). The user can also choose to activate early stopping (<code>--early_stopping</code>), which reduces training time and can help avoiding overfitting.</p>
<div class="codehilite"><pre><span></span><code><span class="n">usage</span><span class="o">:</span> <span class="n">rnasamba</span> <span class="n">train</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">[-</span><span class="n">s</span> <span class="n">EARLY_STOPPING</span><span class="o">]</span> <span class="o">[-</span><span class="n">b</span> <span class="n">BATCH_SIZE</span><span class="o">]</span> <span class="o">[-</span><span class="n">e</span> <span class="n">EPOCHS</span><span class="o">]</span>
                      <span class="o">[-</span><span class="n">v</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">,</span><span class="mi">2</span><span class="o">,</span><span class="mi">3</span><span class="o">}]</span>
                      <span class="n">output_file</span> <span class="n">coding_file</span> <span class="n">noncoding_file</span>

<span class="n">Train</span> <span class="n">a</span> <span class="k">new</span> <span class="n">classification</span> <span class="n">model</span><span class="o">.</span>

<span class="n">positional</span> <span class="k">arguments</span><span class="o">:</span>
  <span class="n">output_file</span>           <span class="n">output</span> <span class="n">HDF5</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">weights</span> <span class="n">of</span> <span class="n">the</span> <span class="n">newly</span>
                        <span class="n">trained</span> <span class="n">RNAsamba</span> <span class="n">network</span><span class="o">.</span>
  <span class="n">coding_file</span>           <span class="n">input</span> <span class="n">FASTA</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">sequences</span> <span class="n">of</span> <span class="n">protein</span><span class="o">-</span>
                        <span class="n">coding</span> <span class="n">transcripts</span><span class="o">.</span>
  <span class="n">noncoding_file</span>        <span class="n">input</span> <span class="n">FASTA</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">sequences</span> <span class="n">of</span> <span class="n">noncoding</span>
                        <span class="n">transcripts</span><span class="o">.</span>

<span class="n">optional</span> <span class="k">arguments</span><span class="o">:</span>
  <span class="o">-</span><span class="n">h</span><span class="o">,</span> <span class="o">--</span><span class="n">help</span>            <span class="n">show</span> <span class="k">this</span> <span class="n">help</span> <span class="n">message</span> <span class="n">and</span> <span class="n">exit</span>
  <span class="o">--</span><span class="n">version</span>             <span class="n">show</span> <span class="n">program</span><span class="err">&#39;</span><span class="n">s</span> <span class="n">version</span> <span class="n">number</span> <span class="n">and</span> <span class="n">exit</span>
  <span class="o">-</span><span class="n">s</span> <span class="n">EARLY_STOPPING</span><span class="o">,</span> <span class="o">--</span><span class="n">early_stopping</span> <span class="n">EARLY_STOPPING</span>
                        <span class="n">number</span> <span class="n">of</span> <span class="n">epochs</span> <span class="n">after</span> <span class="n">lowest</span> <span class="n">validation</span> <span class="n">loss</span> <span class="n">before</span>
                        <span class="n">stopping</span> <span class="n">training</span> <span class="o">(</span><span class="n">a</span> <span class="n">fraction</span> <span class="n">of</span> <span class="mf">0.1</span> <span class="n">of</span> <span class="n">the</span> <span class="n">training</span>
                        <span class="kd">set</span> <span class="k">is</span> <span class="kd">set</span> <span class="n">apart</span> <span class="k">for</span> <span class="n">validation</span> <span class="n">and</span> <span class="n">the</span> <span class="n">model</span> <span class="k">with</span> <span class="n">the</span>
                        <span class="n">lowest</span> <span class="n">validation</span> <span class="n">loss</span> <span class="n">will</span> <span class="n">be</span> <span class="n">saved</span><span class="o">).</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="mi">0</span><span class="o">)</span>
  <span class="o">-</span><span class="n">b</span> <span class="n">BATCH_SIZE</span><span class="o">,</span> <span class="o">--</span><span class="n">batch_size</span> <span class="n">BATCH_SIZE</span>
                        <span class="n">number</span> <span class="n">of</span> <span class="n">samples</span> <span class="n">per</span> <span class="n">gradient</span> <span class="n">update</span><span class="o">.</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="mi">128</span><span class="o">)</span>
  <span class="o">-</span><span class="n">e</span> <span class="n">EPOCHS</span><span class="o">,</span> <span class="o">--</span><span class="n">epochs</span> <span class="n">EPOCHS</span>
                        <span class="n">number</span> <span class="n">of</span> <span class="n">epochs</span> <span class="n">to</span> <span class="n">train</span> <span class="n">the</span> <span class="n">model</span><span class="o">.</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="mi">40</span><span class="o">)</span>
  <span class="o">-</span><span class="n">v</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">,</span><span class="mi">2</span><span class="o">,</span><span class="mi">3</span><span class="o">},</span> <span class="o">--</span><span class="n">verbose</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">,</span><span class="mi">2</span><span class="o">,</span><span class="mi">3</span><span class="o">}</span>
                        <span class="n">print</span> <span class="n">the</span> <span class="n">progress</span> <span class="n">of</span> <span class="n">the</span> <span class="n">training</span><span class="o">.</span> <span class="mi">0</span> <span class="o">=</span> <span class="n">silent</span><span class="o">,</span> <span class="mi">1</span> <span class="o">=</span>
                        <span class="n">current</span> <span class="n">step</span><span class="o">,</span> <span class="mi">2</span> <span class="o">=</span> <span class="n">progress</span> <span class="n">bar</span><span class="o">,</span> <span class="mi">3</span> <span class="o">=</span> <span class="n">one</span> <span class="n">line</span> <span class="n">per</span>
                        <span class="n">epoch</span><span class="o">.</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="mi">0</span><span class="o">)</span>
</code></pre></div>


<h2 id="rnasamba-classify"><code>rnasamba classify</code></h2>
<p><code>rnasamba classify</code> is the command for computing the coding potential of transcripts contained in an input FASTA file and classifying them into coding or non-coding. Optionally, the user can specify an output FASTA file (<code>--protein_fasta</code>) in which RNAsamba will write the translated sequences of the predicted coding ORFs. If multiple weight files are provided, RNAsamba will ensemble their predictions into a single output.</p>
<div class="codehilite"><pre><span></span><code><span class="n">usage</span><span class="o">:</span> <span class="n">rnasamba</span> <span class="n">classify</span> <span class="o">[-</span><span class="n">h</span><span class="o">]</span> <span class="o">[-</span><span class="n">p</span> <span class="n">PROTEIN_FASTA</span><span class="o">]</span> <span class="o">[-</span><span class="n">v</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">}]</span>
                         <span class="n">output_file</span> <span class="n">fasta_file</span> <span class="n">weights</span> <span class="o">[</span><span class="n">weights</span> <span class="o">...]</span>

<span class="n">Classify</span> <span class="n">sequences</span> <span class="n">from</span> <span class="n">a</span> <span class="n">input</span> <span class="n">FASTA</span> <span class="n">file</span><span class="o">.</span>

<span class="n">positional</span> <span class="k">arguments</span><span class="o">:</span>
  <span class="n">output_file</span>           <span class="n">output</span> <span class="n">TSV</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">the</span> <span class="n">results</span> <span class="n">of</span> <span class="n">the</span>
                        <span class="n">classification</span><span class="o">.</span>
  <span class="n">fasta_file</span>            <span class="n">input</span> <span class="n">FASTA</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">transcript</span> <span class="n">sequences</span><span class="o">.</span>
  <span class="n">weights</span>               <span class="n">input</span> <span class="n">HDF5</span> <span class="n">file</span><span class="o">(</span><span class="n">s</span><span class="o">)</span> <span class="n">containing</span> <span class="n">weights</span> <span class="n">of</span> <span class="n">a</span> <span class="n">trained</span>
                        <span class="n">RNAsamba</span> <span class="n">network</span> <span class="o">(</span><span class="k">if</span> <span class="n">more</span> <span class="n">than</span> <span class="n">a</span> <span class="n">file</span> <span class="k">is</span> <span class="n">provided</span><span class="o">,</span> <span class="n">an</span>
                        <span class="n">ensembling</span> <span class="n">of</span> <span class="n">the</span> <span class="n">models</span> <span class="n">will</span> <span class="n">be</span> <span class="n">performed</span><span class="o">).</span>

<span class="n">optional</span> <span class="k">arguments</span><span class="o">:</span>
  <span class="o">-</span><span class="n">h</span><span class="o">,</span> <span class="o">--</span><span class="n">help</span>            <span class="n">show</span> <span class="k">this</span> <span class="n">help</span> <span class="n">message</span> <span class="n">and</span> <span class="n">exit</span>
  <span class="o">--</span><span class="n">version</span>             <span class="n">show</span> <span class="n">program</span><span class="err">&#39;</span><span class="n">s</span> <span class="n">version</span> <span class="n">number</span> <span class="n">and</span> <span class="n">exit</span>
  <span class="o">-</span><span class="n">p</span> <span class="n">PROTEIN_FASTA</span><span class="o">,</span> <span class="o">--</span><span class="n">protein_fasta</span> <span class="n">PROTEIN_FASTA</span>
                        <span class="n">output</span> <span class="n">FASTA</span> <span class="n">file</span> <span class="n">containing</span> <span class="n">translated</span> <span class="n">sequences</span> <span class="k">for</span>
                        <span class="n">the</span> <span class="n">predicted</span> <span class="n">coding</span> <span class="n">ORFs</span><span class="o">.</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="n">None</span><span class="o">)</span>
  <span class="o">-</span><span class="n">v</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">},</span> <span class="o">--</span><span class="n">verbose</span> <span class="o">{</span><span class="mi">0</span><span class="o">,</span><span class="mi">1</span><span class="o">}</span>
                        <span class="n">print</span> <span class="n">the</span> <span class="n">progress</span> <span class="n">of</span> <span class="n">the</span> <span class="n">classification</span><span class="o">.</span> <span class="mi">0</span> <span class="o">=</span> <span class="n">silent</span><span class="o">,</span>
                        <span class="mi">1</span> <span class="o">=</span> <span class="n">current</span> <span class="n">step</span><span class="o">.</span> <span class="o">(</span><span class="k">default</span><span class="o">:</span> <span class="mi">0</span><span class="o">)</span>
</code></pre></div>


<h2 id="examples">Examples</h2>
<ul>
<li>Training a new classification model using <em>Mus musculus</em> data downloaded from GENCODE:</li>
</ul>
<div class="codehilite"><pre><span></span><code><span class="err">rnasamba train -v 2 mouse_model.hdf5 gencode.vM21.pc_transcripts.fa gencode.vM21.lncRNA_transcripts.fa</span>
</code></pre></div>


<ul>
<li>Classifying sequences using our pre-trained model (<code>partial_length_weights.hdf5</code>) and saving the predicted proteins into a FASTA file:</li>
</ul>
<div class="codehilite"><pre><span></span><code><span class="err">rnasamba classify -p predicted_proteins.fa classification.tsv input.fa partial_length_weights.hdf5</span>
</code></pre></div>


<div class="codehilite"><pre><span></span><code><span class="n">head</span> <span class="n">classification</span><span class="p">.</span><span class="n">tsv</span>

<span class="n">sequence_name</span>   <span class="n">coding_score</span>    <span class="n">classification</span>
<span class="n">ENSMUST00000054910</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">99022</span> <span class="n">coding</span>
<span class="n">ENSMUST00000059648</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">84718</span> <span class="n">coding</span>
<span class="n">ENSMUST00000055537</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">99713</span> <span class="n">coding</span>
<span class="n">ENSMUST00000030975</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">85189</span> <span class="n">coding</span>
<span class="n">ENSMUST00000050754</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">02638</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000008011</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">14949</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000061643</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">03456</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000059704</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">89232</span> <span class="n">coding</span>
<span class="n">ENSMUST00000036304</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">03782</span> <span class="n">noncoding</span>
</code></pre></div>


<h2 id="using-the-docker-image">Using the Docker image</h2>
<div class="codehilite"><pre><span></span><code><span class="o">#</span> <span class="n">Pull</span> <span class="n">the</span> <span class="n">Docker</span> <span class="n">image</span> <span class="k">to</span> <span class="n">your</span> <span class="n">computer</span><span class="p">:</span>
<span class="n">docker</span> <span class="n">pull</span> <span class="n">antoniopcamargo</span><span class="o">/</span><span class="n">rnasamba</span>

<span class="o">#</span> <span class="n">Training</span> <span class="n">example</span><span class="p">:</span>
<span class="n">docker</span> <span class="n">run</span> <span class="o">-</span><span class="n">ti</span> <span class="c1">--rm -v &quot;$(pwd):/app&quot; antoniopcamargo/rnasamba train -v 2 mouse_model.hdf5 gencode.vM21.pc_transcripts.fa gencode.vM21.lncRNA_transcripts.fa</span>

<span class="o">#</span> <span class="n">Classification</span> <span class="n">example</span><span class="p">:</span>
<span class="n">docker</span> <span class="n">run</span> <span class="o">-</span><span class="n">ti</span> <span class="c1">--rm -v &quot;$(pwd):/app&quot; antoniopcamargo/rnasamba classify -p predicted_proteins.fa classification.tsv input.fa full_length_weights.hdf5</span>
</code></pre></div>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../theory/" title="Theory" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Theory
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="https://www.mkdocs.org" target="_blank" rel="noopener">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs</a>
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../assets/fonts/font-awesome.css">
    
      <a href="https://github.com/apcamargo/" target="_blank" rel="noopener" title="github-alt" class="md-footer-social__link fa fa-github-alt"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application.c33a9706.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:".."}})</script>
      
    
  </body>
</html>