<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="description" content="Documentation for RNAsamba, a coding potential assessment tool"><link rel="canonical" href="https://apcamargo.github.io/RNAsamba/usage/"><meta name="author" content="Antonio Camargo"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.0"><title>Usage - RNAsamba Documentation</title><link rel="stylesheet" href="../assets/stylesheets/application.4031d38b.css"><link rel="stylesheet" href="../assets/stylesheets/application-palette.224b79ff.css"><meta name="theme-color" content="#ef5350"><script src="../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"IBM Plex Sans","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../assets/fonts/material-icons.css"><link rel="stylesheet" href="../extra.css"></head><body dir="ltr" data-md-color-primary="red" data-md-color-accent="red"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#usage" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="https://apcamargo.github.io/RNAsamba/" title="RNAsamba Documentation" class="md-header-nav__button md-logo"><i class="md-icon">book</i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">RNAsamba Documentation</span><span class="md-header-nav__topic">Usage</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/apcamargo/RNAsamba/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">apcamargo/RNAsamba</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="https://apcamargo.github.io/RNAsamba/" title="RNAsamba Documentation" class="md-nav__button md-logo"><i class="md-icon">book</i></a>RNAsamba Documentation</label><div class="md-nav__source"><a href="https://github.com/apcamargo/RNAsamba/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">apcamargo/RNAsamba</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href=".." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item"><a href="../theory/" title="Theory" class="md-nav__link">Theory</a></li><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Usage</label><a href="./" title="Usage" class="md-nav__link md-nav__link--active">Usage</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#download-the-pre-trained-models" title="Download the pre-trained models" class="md-nav__link">Download the pre-trained models</a></li><li class="md-nav__item"><a href="#rnasamba-train" title="rnasamba-train" class="md-nav__link">rnasamba-train</a></li><li class="md-nav__item"><a href="#rnasamba-classify" title="rnasamba-classify" class="md-nav__link">rnasamba-classify</a></li><li class="md-nav__item"><a href="#examples" title="Examples" class="md-nav__link">Examples</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#download-the-pre-trained-models" title="Download the pre-trained models" class="md-nav__link">Download the pre-trained models</a></li><li class="md-nav__item"><a href="#rnasamba-train" title="rnasamba-train" class="md-nav__link">rnasamba-train</a></li><li class="md-nav__item"><a href="#rnasamba-classify" title="rnasamba-classify" class="md-nav__link">rnasamba-classify</a></li><li class="md-nav__item"><a href="#examples" title="Examples" class="md-nav__link">Examples</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/apcamargo/RNAsamba/edit/master/docs/usage.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><h1 id="usage">Usage</h1>
<h2 id="download-the-pre-trained-models">Download the pre-trained models</h2>
<p>We provide two HDF5 files containing the weights of classification models trained with human trancript sequences. The first model (<code>full_length_weights.hdf5</code>) was trained exclusively with full-length transcripts and can be used in datasets comprised mostly or exclusively of complete transcript sequences. The second model (<code>partial_length_weights.hdf5</code>) was trained with both complete and truncated transcripts and is prefered in cases where there is a significant fraction of partial-length sequences, such as transcriptomes assembled using <em>de novo</em> approaches.</p>
<p><center>
  <button onclick="location.href='https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/full_length_weights.hdf5'" class="pure-material-button-contained">Full-length transcripts</button>
  <button onclick="location.href='https://raw.githubusercontent.com/apcamargo/RNAsamba/master/data/partial_length_weights.hdf5'" class="pure-material-button-contained">Partial-length transcripts</button>
</center></p>
<p>Alternatively, you can download the weight files in an command-line interface by executing the following commands:</p>
<div class="codehilite"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">O</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="p">.</span><span class="n">githubusercontent</span><span class="p">.</span><span class="n">com</span><span class="o">/</span><span class="n">apcamargo</span><span class="o">/</span><span class="n">RNAsamba</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="k">data</span><span class="o">/</span><span class="n">full_length_weights</span><span class="p">.</span><span class="n">hdf5</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">O</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="p">.</span><span class="n">githubusercontent</span><span class="p">.</span><span class="n">com</span><span class="o">/</span><span class="n">apcamargo</span><span class="o">/</span><span class="n">RNAsamba</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="k">data</span><span class="o">/</span><span class="n">partial_length_weights</span><span class="p">.</span><span class="n">hdf5</span>
</pre></div>


<p>Both models achieves high classification performance in transcripts from a variety of different species (see our <a href="https://www.biorxiv.org/content/10.1101/620880v1">article</a>).</p>
<div class="admonition warning">
<p>In case you want to train your own model, you should follow the steps described in the <a href="#rnasamba-train"><code>rnasamba-train</code></a> section.</p>
</div>
<h2 id="rnasamba-train"><code>rnasamba-train</code></h2>
<p><code>rnasamba-train</code> is the command for training a new classification model from a training dataset and saving the network weights into an HDF5 file. The user can specify the batch size (<code>--batch_size</code>) and the number of training epochs (<code>--epochs</code>). The user can also choose to activate early stopping (<code>--early_stopping</code>), which reduces training time and can help avoiding overfitting.</p>
<div class="codehilite"><pre><span></span><span class="nv">usage</span>: <span class="nv">rnasamba</span><span class="o">-</span><span class="nv">train</span> [<span class="o">-</span><span class="nv">h</span>] [<span class="o">-</span><span class="nv">s</span> <span class="nv">EARLY_STOPPING</span>] [<span class="o">-</span><span class="nv">b</span> <span class="nv">BATCH_SIZE</span>] [<span class="o">-</span><span class="nv">e</span> <span class="nv">EPOCHS</span>]
                      [<span class="o">-</span><span class="nv">v</span> {<span class="mi">0</span>,<span class="mi">1</span>,<span class="mi">2</span>,<span class="mi">3</span>}]
                      <span class="nv">output_file</span> <span class="nv">coding_file</span> <span class="nv">noncoding_file</span>

<span class="nv">Train</span> <span class="nv">a</span> <span class="nv">new</span> <span class="nv">classification</span> <span class="nv">model</span>.

<span class="nv">positional</span> <span class="nv">arguments</span>:
  <span class="nv">output_file</span>           <span class="nv">output</span> <span class="nv">HDF5</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">weights</span> <span class="nv">of</span> <span class="nv">the</span> <span class="nv">newly</span>
                        <span class="nv">trained</span> <span class="nv">RNAsamba</span> <span class="nv">network</span>.
  <span class="nv">coding_file</span>           <span class="nv">input</span> <span class="nv">FASTA</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">sequences</span> <span class="nv">of</span> <span class="nv">protein</span><span class="o">-</span>
                        <span class="nv">coding</span> <span class="nv">transcripts</span>.
  <span class="nv">noncoding_file</span>        <span class="nv">input</span> <span class="nv">FASTA</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">sequences</span> <span class="nv">of</span> <span class="nv">noncoding</span>
                        <span class="nv">transcripts</span>.

<span class="nv">optional</span> <span class="nv">arguments</span>:
  <span class="o">-</span><span class="nv">h</span>, <span class="o">--</span><span class="nv">help</span>            <span class="k">show</span> <span class="nv">this</span> <span class="nv">help</span> <span class="nv">message</span> <span class="nv">and</span> <span class="k">exit</span>
  <span class="o">-</span><span class="nv">s</span> <span class="nv">EARLY_STOPPING</span>, <span class="o">--</span><span class="nv">early_stopping</span> <span class="nv">EARLY_STOPPING</span>
                        <span class="nv">number</span> <span class="nv">of</span> <span class="nv">epochs</span> <span class="nv">after</span> <span class="nv">lowest</span> <span class="nv">validation</span> <span class="nv">loss</span> <span class="nv">before</span>
                        <span class="nv">stopping</span> <span class="nv">training</span> <span class="ss">(</span><span class="nv">a</span> <span class="nv">fraction</span> <span class="nv">of</span> <span class="mi">0</span>.<span class="mi">1</span> <span class="nv">of</span> <span class="nv">the</span> <span class="nv">training</span>
                        <span class="nv">set</span> <span class="nv">is</span> <span class="nv">set</span> <span class="nv">apart</span> <span class="k">for</span> <span class="nv">validation</span> <span class="nv">and</span> <span class="nv">the</span> <span class="nv">model</span> <span class="nv">with</span> <span class="nv">the</span>
                        <span class="nv">lowest</span> <span class="nv">validation</span> <span class="nv">loss</span> <span class="nv">will</span> <span class="nv">be</span> <span class="nv">saved</span><span class="ss">)</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="mi">0</span><span class="ss">)</span>
  <span class="o">-</span><span class="nv">b</span> <span class="nv">BATCH_SIZE</span>, <span class="o">--</span><span class="nv">batch_size</span> <span class="nv">BATCH_SIZE</span>
                        <span class="nv">number</span> <span class="nv">of</span> <span class="nv">samples</span> <span class="nv">per</span> <span class="nv">gradient</span> <span class="nv">update</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="mi">128</span><span class="ss">)</span>
  <span class="o">-</span><span class="nv">e</span> <span class="nv">EPOCHS</span>, <span class="o">--</span><span class="nv">epochs</span> <span class="nv">EPOCHS</span>
                        <span class="nv">number</span> <span class="nv">of</span> <span class="nv">epochs</span> <span class="nv">to</span> <span class="nv">train</span> <span class="nv">the</span> <span class="nv">model</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="mi">40</span><span class="ss">)</span>
  <span class="o">-</span><span class="nv">v</span> {<span class="mi">0</span>,<span class="mi">1</span>,<span class="mi">2</span>,<span class="mi">3</span>}, <span class="o">--</span><span class="nv">verbose</span> {<span class="mi">0</span>,<span class="mi">1</span>,<span class="mi">2</span>,<span class="mi">3</span>}
                        <span class="nv">print</span> <span class="nv">the</span> <span class="nv">progress</span> <span class="nv">of</span> <span class="nv">the</span> <span class="nv">training</span>. <span class="mi">0</span> <span class="o">=</span> <span class="nv">silent</span>, <span class="mi">1</span> <span class="o">=</span>
                        <span class="nv">current</span> <span class="nv">step</span>, <span class="mi">2</span> <span class="o">=</span> <span class="nv">progress</span> <span class="nv">bar</span>, <span class="mi">3</span> <span class="o">=</span> <span class="nv">one</span> <span class="nv">line</span> <span class="nv">per</span>
                        <span class="nv">epoch</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="mi">0</span><span class="ss">)</span>
</pre></div>


<h2 id="rnasamba-classify"><code>rnasamba-classify</code></h2>
<p><code>rnasamba-classify</code> is the command for computing the coding potential of transcripts contained in an input FASTA file and classifying them into coding or non-coding. Optionally, the user can specify an output FASTA file (<code>--protein_fasta</code>) in which RNAsamba will write the translated sequences of the predicted coding ORFs. If multiple weight files are provided, RNAsamba will ensemble their predictions into a single output.</p>
<div class="codehilite"><pre><span></span><span class="nv">usage</span>: <span class="nv">rnasamba</span><span class="o">-</span><span class="nv">classify</span> [<span class="o">-</span><span class="nv">h</span>] [<span class="o">-</span><span class="nv">p</span> <span class="nv">PROTEIN_FASTA</span>] [<span class="o">-</span><span class="nv">v</span> {<span class="mi">0</span>,<span class="mi">1</span>}]
                         <span class="nv">output_file</span> <span class="nv">fasta_file</span> <span class="nv">weights</span> [<span class="nv">weights</span> ...]

<span class="nv">Classify</span> <span class="nv">sequences</span> <span class="nv">from</span> <span class="nv">a</span> <span class="nv">input</span> <span class="nv">FASTA</span> <span class="nv">file</span>.

<span class="nv">positional</span> <span class="nv">arguments</span>:
  <span class="nv">output_file</span>           <span class="nv">output</span> <span class="nv">TSV</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">the</span> <span class="nv">results</span> <span class="nv">of</span> <span class="nv">the</span>
                        <span class="nv">classification</span>.
  <span class="nv">fasta_file</span>            <span class="nv">input</span> <span class="nv">FASTA</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">transcript</span> <span class="nv">sequences</span>.
  <span class="nv">weights</span>               <span class="nv">input</span> <span class="nv">HDF5</span> <span class="nv">file</span><span class="ss">(</span><span class="nv">s</span><span class="ss">)</span> <span class="nv">containing</span> <span class="nv">weights</span> <span class="nv">of</span> <span class="nv">a</span> <span class="nv">trained</span>
                        <span class="nv">RNAsamba</span> <span class="nv">network</span> <span class="ss">(</span><span class="k">if</span> <span class="nv">more</span> <span class="nv">than</span> <span class="nv">a</span> <span class="nv">file</span> <span class="nv">is</span> <span class="nv">provided</span>, <span class="nv">an</span>
                        <span class="nv">ensembling</span> <span class="nv">of</span> <span class="nv">the</span> <span class="nv">models</span> <span class="nv">will</span> <span class="nv">be</span> <span class="nv">performed</span><span class="ss">)</span>.

<span class="nv">optional</span> <span class="nv">arguments</span>:
  <span class="o">-</span><span class="nv">h</span>, <span class="o">--</span><span class="nv">help</span>            <span class="k">show</span> <span class="nv">this</span> <span class="nv">help</span> <span class="nv">message</span> <span class="nv">and</span> <span class="k">exit</span>
  <span class="o">-</span><span class="nv">p</span> <span class="nv">PROTEIN_FASTA</span>, <span class="o">--</span><span class="nv">protein_fasta</span> <span class="nv">PROTEIN_FASTA</span>
                        <span class="nv">output</span> <span class="nv">FASTA</span> <span class="nv">file</span> <span class="nv">containing</span> <span class="nv">translated</span> <span class="nv">sequences</span> <span class="k">for</span>
                        <span class="nv">the</span> <span class="nv">predicted</span> <span class="nv">coding</span> <span class="nv">ORFs</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="nv">None</span><span class="ss">)</span>
  <span class="o">-</span><span class="nv">v</span> {<span class="mi">0</span>,<span class="mi">1</span>}, <span class="o">--</span><span class="nv">verbose</span> {<span class="mi">0</span>,<span class="mi">1</span>}
                        <span class="nv">print</span> <span class="nv">the</span> <span class="nv">progress</span> <span class="nv">of</span> <span class="nv">the</span> <span class="nv">classification</span>. <span class="mi">0</span> <span class="o">=</span> <span class="nv">silent</span>,
                        <span class="mi">1</span> <span class="o">=</span> <span class="nv">current</span> <span class="nv">step</span>. <span class="ss">(</span><span class="nv">default</span>: <span class="mi">0</span><span class="ss">)</span>
</pre></div>


<h2 id="examples">Examples</h2>
<ul>
<li>Training a new classification model using <em>Mus musculus</em> data downloaded from GENCODE:</li>
</ul>
<div class="codehilite"><pre><span></span><span class="n">rnasamba</span><span class="o">-</span><span class="n">train</span> <span class="n">mouse_model</span><span class="p">.</span><span class="n">hdf5</span> <span class="o">-</span><span class="n">v</span> <span class="mi">2</span> <span class="n">gencode</span><span class="p">.</span><span class="n">vM21</span><span class="p">.</span><span class="n">pc_transcripts</span><span class="p">.</span><span class="n">fa</span> <span class="n">gencode</span><span class="p">.</span><span class="n">vM21</span><span class="p">.</span><span class="n">lncRNA_transcripts</span><span class="p">.</span><span class="n">fa</span>
</pre></div>


<ul>
<li>Classifying sequences using our pre-trained model (<code>full_length_weights.hdf5</code>) and saving the predicted proteins into a FASTA file:</li>
</ul>
<div class="codehilite"><pre><span></span><span class="n">rnasamba</span><span class="o">-</span><span class="n">classify</span> <span class="o">-</span><span class="n">p</span> <span class="n">predicted_proteins</span><span class="p">.</span><span class="n">fa</span> <span class="n">classification</span><span class="p">.</span><span class="n">tsv</span> <span class="k">input</span><span class="p">.</span><span class="n">fa</span> <span class="n">full_length_weights</span><span class="p">.</span><span class="n">hdf5</span>
</pre></div>


<div class="codehilite"><pre><span></span><span class="n">head</span> <span class="n">classification</span><span class="p">.</span><span class="n">tsv</span>

<span class="n">sequence_name</span>   <span class="n">coding_score</span>    <span class="n">classification</span>
<span class="n">ENSMUST00000054910</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">99022</span> <span class="n">coding</span>
<span class="n">ENSMUST00000059648</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">84718</span> <span class="n">coding</span>
<span class="n">ENSMUST00000055537</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">99713</span> <span class="n">coding</span>
<span class="n">ENSMUST00000030975</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">85189</span> <span class="n">coding</span>
<span class="n">ENSMUST00000050754</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">02638</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000008011</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">14949</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000061643</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">03456</span> <span class="n">noncoding</span>
<span class="n">ENSMUST00000059704</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">89232</span> <span class="n">coding</span>
<span class="n">ENSMUST00000036304</span>  <span class="mi">0</span><span class="p">.</span><span class="mi">03782</span> <span class="n">noncoding</span>
</pre></div></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../theory/" title="Theory" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Theory</span></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright">powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div><div class="md-footer-social"><link rel="stylesheet" href="../assets/fonts/font-awesome.css"> <a href="https://github.com/apcamargo/" class="md-footer-social__link fa fa-github-alt"></a> </div></div></div></footer></div><script src="../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:".."}})</script></body></html>